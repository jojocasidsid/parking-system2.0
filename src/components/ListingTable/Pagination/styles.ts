import styled from '@emotion/styled'
import { Typography } from '@mui/material'

export interface IPageInputProps {
	error?: string
}

export const PageInput = styled.input<IPageInputProps>`
	border-radius: 2px 0px 0px 2px;
	padding: 7px 2px 7px 0px;
	text-align: center;
	width: 100%;
	font-style: normal;
	font-size: 14px;
	line-height: 16px;
	font-weight: 500;
	font-stretch: normal;
	height: 28px;

	&::-webkit-outer-spin-button,
	&::-webkit-inner-spin-button {
		-webkit-appearance: none;
		margin: 0;
	}

	&:active,
	&:focus {
		border: 1px solid ${({ theme }) => theme.palette.primary.main};
		outline: none;
		box-shadow: none;
	}

	&:disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	border: 1px solid
		${({ theme, error }) => {
			if (error) {
				return theme?.palette.error.main
			}
			return theme.palette.text.secondary
		}};
`

export const StyledTypography = styled(Typography)`
	color: #303643b3;
	font-weight: bold;
	font-size: 12px;
`
