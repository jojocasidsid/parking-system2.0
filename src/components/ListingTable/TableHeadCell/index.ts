import TableHeadCell from './TableHeadCell'

export default TableHeadCell
