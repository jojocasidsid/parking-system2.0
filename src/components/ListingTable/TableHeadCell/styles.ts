import styled from '@emotion/styled'
import Typography from '@mui/material/Typography'

export const StyledTableHeadTypography = styled(Typography)`
	font-weight: bold;
	font-size: 14px;
`
