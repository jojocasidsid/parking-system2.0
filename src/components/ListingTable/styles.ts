import styled from '@emotion/styled'
import { Typography } from '@mui/material'

export const StyledTypography = styled(Typography)`
	color: #30364380;
	font-weight: bold;
	font-size: 14px;
`
