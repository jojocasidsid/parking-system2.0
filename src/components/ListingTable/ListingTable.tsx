import * as React from 'react'

// packages
import {
	Box,
	Table,
	TableContainer,
	Typography,
	Grid,
	Divider,
} from '@mui/material'

// components
import LoadingIndicator from 'components/LoadingIndicator'

// table related components
import TableFilters from './TableFilters'
import TableBodyCell from './TableBodyCell'
import TableHeadCell from './TableHeadCell'
import Pagination from './Pagination'

// interface
import { IProps } from './types'

// types
import { StyledTypography } from './styles'

const ListingTable = ({
	title,
	isPagination,
	isFilters,
	filters,
	dense,
	columns,
	rows,
	totalPages,
	isLoading,
	emptyTableMessage,
	setFilters,
}: IProps) => (
	<div aria-labelledby='tableHeading'>
		<Grid container>
			<Grid item xs={12} container padding={2.5}>
				{title && (
					<Grid item paddingBottom={3.5}>
						<Typography id='tableHeading' variant='h1' fontWeight='bold'>
							{title}
						</Typography>
					</Grid>
				)}
				<Grid item container columnSpacing={1.5}>
					{isFilters && (
						<TableFilters filters={filters && filters} setFilters={setFilters} />
					)}
				</Grid>
			</Grid>
			<Box width='100%'>
				<Divider style={{ opacity: 0.3 }} />
			</Box>
			<TableContainer>
				<Table aria-labelledby='tableTitle' size={dense || 'medium'} stickyHeader>
					<TableHeadCell
						isFilters={isFilters}
						filters={filters}
						setFilters={setFilters}
						columns={columns}
					/>
					{!isLoading && <TableBodyCell columns={columns} rows={rows} />}
				</Table>
			</TableContainer>
			{isLoading && (
				<Grid container justifyContent='center' padding={2}>
					<Grid item>
						<LoadingIndicator />
					</Grid>
				</Grid>
			)}
			{rows.length === 0 && !isLoading && (
				<Grid container justifyContent='center' padding={2}>
					<Grid item>
						<StyledTypography>{emptyTableMessage}</StyledTypography>
					</Grid>
				</Grid>
			)}
			{isPagination && (
				<Pagination
					filters={filters}
					setFilters={setFilters}
					totalPages={totalPages}
				/>
			)}
		</Grid>
	</div>
)

export default ListingTable
