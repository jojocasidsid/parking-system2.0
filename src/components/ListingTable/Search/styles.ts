import styled from '@emotion/styled'
import { InputLabel, TextField } from '@mui/material'
import { Theme } from '@mui/material/styles'

interface IInput {
	error?: boolean
	theme?: Theme
	small?: boolean
}

export const StyledRoot = styled('div')`
	display: block;
`

export const StyledLabel = styled(InputLabel)`
	text-align: 'left';
	letter-spacing: 0.12px;
	padding-bottom: 8px;

	font-style: normal;
	font-size: 12px;
	line-height: 14px;
	font-weight: 600;
	font-stretch: normal;
	white-space: normal;
	//has error
	color: ${({ theme, error }: IInput) => {
		if (error) {
			return theme?.palette.error.main
		}
		return theme?.palette.text.primary
	}};
	& span {
		color: ${({ theme }) => theme.palette.error.main};
	}

	display: block;
`

export const StyledInput = styled(TextField)`
	input {
		padding: ${({ small }: IInput) =>
			small ? '2px 8px 2px 8px' : '8px 12px 8px 16px'};
		font-style: normal;
		font-size: 14px;
		font-weight: 500;
		font-stretch: normal;
		color: ${({ theme }) => theme?.palette.text.primary};
	}

	fieldset {
		border: 1px solid ${({ theme }) => theme.palette.text.secondary};
		border-radius: 2px 0px 0px 2px;
	}

	input.Mui-disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	input.MuiInputBase-inputAdornedStart {
		padding-left: 0px !important;
	}

	input.MuiInputBase-inputAdornedEnd {
		padding-right: 0px;
	}

	.MuiInputBase-adornedEnd div.MuiInputAdornment-root {
		background-color: ${({ theme }) => theme.palette.primary.contrastText};
		padding: 16px 16px 16px 16px;
		padding-right: 0px !important;
	}

	.MuiInputBase-adornedStart {
		padding-left: 0px !important;
	}

	.MuiInputBase-adornedStart .MuiInputAdornment-root {
		background-color: ${({ theme }) => theme.palette.primary.contrastText};
		padding: 16px 16px 16px 16px;
	}

	.MuiFormHelperText-root {
		margin-left: 0;
	}
`
