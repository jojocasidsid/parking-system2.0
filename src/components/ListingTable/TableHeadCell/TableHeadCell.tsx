import React from 'react'

import {
	Box,
	TableHead,
	TableRow,
	TableSortLabel,
	TableCell,
} from '@mui/material'

import { visuallyHidden } from '@mui/utils'
import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown'
import styled from '@emotion/styled'

import { StyledTableHeadTypography } from './styles'
import { IColumns, IFilters, IRows } from '../types'

const StyledTableCell = styled(TableCell)`
	white-space: nowrap;
`

interface IProps {
	isFilters?: boolean
	filters?: IFilters
	setFilters?: React.Dispatch<React.SetStateAction<IFilters>>
	columns: IColumns<IRows>[]
}

const TableHeadCell = ({ isFilters, filters, columns, setFilters }: IProps) => {
	const sortIdentifier = (sort: 'asc' | 'desc') => {
		if (sort === 'asc') {
			return 'desc'
		}
		if (sort === 'desc') {
			return 'asc'
		}
		return 'asc'
	}

	return (
		<TableHead>
			<TableRow>
				{columns.map((column) =>
					isFilters && filters ? (
						<StyledTableCell
							style={{ whiteSpace: 'nowrap' }}
							key={column.id}
							sortDirection={filters._sort === column.id ? filters._order : false}
						>
							{column.sortable ? (
								<TableSortLabel
									active={filters._sort === column.id ? true : undefined}
									direction={filters._order}
									onClick={() =>
										setFilters &&
										setFilters(
											(prevState: IFilters): IFilters => ({
												...prevState,
												_sort: column.id,
												_order: sortIdentifier(filters._order ? filters._order : 'asc'),
											})
										)
									}
									IconComponent={ArrowDropDownIcon}
								>
									<StyledTableHeadTypography>{column.label}</StyledTableHeadTypography>
									{filters._sort === column.id ? (
										<Box component='span' sx={visuallyHidden}>
											{filters._sort === 'desc' ? 'sorted descending' : 'sorted ascending'}
										</Box>
									) : null}
								</TableSortLabel>
							) : (
								<StyledTableHeadTypography>{column.label}</StyledTableHeadTypography>
							)}
						</StyledTableCell>
					) : (
						<StyledTableCell key={column.id}>
							<StyledTableHeadTypography>{column.label}</StyledTableHeadTypography>
						</StyledTableCell>
					)
				)}
			</TableRow>
		</TableHead>
	)
}

export default TableHeadCell
