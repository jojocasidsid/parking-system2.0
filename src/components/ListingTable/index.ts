import ListingTable from './ListingTable'

export default ListingTable
