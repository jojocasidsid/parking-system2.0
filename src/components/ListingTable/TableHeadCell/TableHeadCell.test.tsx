import React, { useState } from 'react'
import renderer from 'react-test-renderer'

// packages
import { ThemeProvider } from '@mui/material/styles'

// config
import theme from 'config/theme'

// component
import { IColumns, IRows } from 'components/ListingTable/types'
import { IFilters } from 'hooks/useListingTable'
import TableHeadCell from './TableHeadCell'

export const columns: IColumns<IRows>[] = [
	{
		id: 'id',
		label: 'ID',
		sortable: true,
	},
	{
		id: 'vehicle',
		label: 'Plate Number',
		sortable: false,
	},
	{
		id: 'parkingType',
		label: 'Parking Type',
		sortable: true,
	},

	{
		id: 'hours',
		label: 'Hours Spent',
		sortable: true,
	},

	{
		id: 'price',
		label: 'Price',
		sortable: false,
	},

	{
		id: 'transactionDate',
		label: 'Transaction Date',
		sortable: true,
	},
]

describe('Table Body Cell  Component', () => {
	it('should render Table Body Cell component without crashing', () => {
		const Component = () => {
			const [filters, setFilters] = useState<IFilters>({
				_order: 'asc',
				_sort: '',
				_page: 1,
				_limit: 20,
				q: '',
			})

			return (
				<ThemeProvider theme={theme}>
					<TableHeadCell
						isFilters
						filters={filters}
						setFilters={setFilters}
						columns={columns}
					/>
				</ThemeProvider>
			)
		}

		const tree = renderer.create(<Component />)
		expect(tree).toMatchSnapshot()
	})
})
