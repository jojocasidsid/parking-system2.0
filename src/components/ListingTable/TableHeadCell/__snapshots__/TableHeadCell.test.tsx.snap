// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Body Cell  Component should render Table Body Cell component without crashing 1`] = `
<thead
  className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
  role={null}
>
  <tr
    className="MuiTableRow-root MuiTableRow-head css-1fztqu1-MuiTableRow-root"
    role={null}
  >
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1isvuh8-MuiTableCell-root"
      scope="col"
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      <span
        className="MuiButtonBase-root MuiTableSortLabel-root css-1lphovs-MuiButtonBase-root-MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1ydb6uk-MuiTypography-root"
        >
          ID
        </p>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m7 10 5 5 5-5z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1isvuh8-MuiTableCell-root"
      scope="col"
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1ydb6uk-MuiTypography-root"
      >
        Plate Number
      </p>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1isvuh8-MuiTableCell-root"
      scope="col"
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      <span
        className="MuiButtonBase-root MuiTableSortLabel-root css-1lphovs-MuiButtonBase-root-MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1ydb6uk-MuiTypography-root"
        >
          Parking Type
        </p>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m7 10 5 5 5-5z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1isvuh8-MuiTableCell-root"
      scope="col"
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      <span
        className="MuiButtonBase-root MuiTableSortLabel-root css-1lphovs-MuiButtonBase-root-MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1ydb6uk-MuiTypography-root"
        >
          Hours Spent
        </p>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m7 10 5 5 5-5z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1isvuh8-MuiTableCell-root"
      scope="col"
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1ydb6uk-MuiTypography-root"
      >
        Price
      </p>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1isvuh8-MuiTableCell-root"
      scope="col"
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      <span
        className="MuiButtonBase-root MuiTableSortLabel-root css-1lphovs-MuiButtonBase-root-MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1ydb6uk-MuiTypography-root"
        >
          Transaction Date
        </p>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m7 10 5 5 5-5z"
          />
        </svg>
      </span>
    </th>
  </tr>
</thead>
`;
